### DISCUSSION TITLE
How can I fix this npm build error?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#37620] [Help] [answered]
So, in my code I have table and I am using this " symbol in the text which has the meaning of inches.

As in 1 1/4" , meaning 1 1/4 inches

The code looks like this:

```
<td>1/4", 1-1/2"</td>
 <td>1/4", 1-1/4"</td>
```
During npm build I get an error that looks like this:

```
105:26  Error: `"` can be escaped with `&quot;`, `&ldquo;`, `&#34;`, `&rdquo;`.  react/no-unescaped-entities
105:34  Error: `"` can be escaped with `&quot;`, `&ldquo;`, `&#34;`, `&rdquo;`.  react/no-unescaped-entities
```
 What is that exactly telling me to do?

**Comment by icyJoseph** [comment, marked as answer]
This happens because the linter tries to prevent syntax errors in JSX, which is JavaScript, so accidentally `"` or `'` could be escaping strings. If you want to learn more about this lint rule: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md

## Solution

```jsx
<td>1/4&quot;, 1-1/2&quot;</td>
 <td>1/4&quot;, 1-1/4&quot;</td>
```

Or...

```jsx
<td>{'1/4", 1-1/2"'}</td>
 <td>{'1/4", 1-1/4"'}</td>
```

In fact there's a bunch of ways:

```jsx
 <td>{'1/4", 1-1/2"'}</td>
 <td>{'1/4", 1-1/4"'}</td>
 <td>1/4&quot;, 1-1/2&quot;</td>
 <td>1/4&quot;, 1-1/4&quot;</td>
 <td>1/4&Prime;, 1-1/2&Prime;</td>
 <td>1/4&Prime;, 1-1/4&Prime;</td>
```

I think the correct HTML entity is `&Prime;`, when talking about inches anyway. You'll have to google that bit.

Adding a screenshot because the nature of HTML entities makes it so that sometimes they resolve to the symbol (the screenshot also includes the case you have, where the linter complains):

<img width="464" alt="Screenshot 2022-06-10 at 17 28 37" src="https://user-images.githubusercontent.com/21013447/173099608-3283dfa0-6508-44c2-a615-5de1345328ca.png">

**Reply by icyJoseph** [reply]
### More screenshots

#### Code
<img width="684" alt="Screenshot 2022-06-10 at 17 33 58" src="https://user-images.githubusercontent.com/21013447/173100520-db26e777-85c2-402c-a6d5-1958d0cc1915.png">

#### Result

![Screenshot 2022-06-10 at 17 34 16](https://user-images.githubusercontent.com/21013447/173100651-6c3902eb-4b0e-4c9a-833a-bfdb33afda50.png)

It seems to me like the 3rd case, using the `&Prime;` HTML entity is more correct.

**Reply by Elindo586** [reply]
Thanks a million... that is a good reference to keep in mind when using characters in the text.....  In the future  I am gonna need all of these too.. ¡  ¿ ñ ó  etc.....  but that is for later...

**Comment by AnweshGangula** [comment]
You can alternatively disable the eslint rule with below config in the `.eslintrc.json` file

```json
	"rules":{
		"react/no-unescaped-entities": 0
	}
```

check out this [Stack overflow post](https://stackoverflow.com/q/43177074/6908282) for reference

**Reply by ndanyikevin** [reply]
Thanks a million

**Comment by RavenHursT** [comment]
Ok.. but how can we get this to work w/ `next lint --fix`??

Absolutely frustrating that this has to be done by hand, breaks builds by default, but there's no way to just have the linter fix it for you?

**Reply by icyJoseph** [reply]
Well, someone has to go to that lint rule repository, and write a fix rule for it.

Here I think: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib%2Frules%2Fno-unescaped-entities.js

And write something like: https://eslint.org/docs/latest/extend/custom-rules#applying-fixes

Though,

- https://eslint.org/docs/latest/use/command-line-interface#fix-problems

As the link above says, not all errors are fixable.
